agent/cleanup.sh

```diff
@@ -21,3 +21,5 @@ case "${AGENT_E2E_TEST_BOOT_MODE}" in
     rm -rf ${WORKING_DIR}/pxe
     ;;
 esac
+
+sudo systemctl stop haproxy
```

Comment:
This requires an `|| true` to avoid job failure in case `haproxy` wasn't started

Reply:
Also, by using the following approach, `/etc/haproxy/haproxy.cfg` should be deleted

Reply:
Thanks. Will update.